systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/externalClients/OauthKafkaClient.java

```diff
@@ -4,29 +4,257 @@
+
+        String caCertName = this.caCertName == null ?
+                KafkaResource.getKafkaExternalListenerCaCertName(namespace, clusterName) : this.caCertName;
+        LOGGER.info("Going to use the following CA certificate: {}", caCertName);
+
+        vertx.deployVerticle(new Producer(KafkaClientProperties.createOauthProducerTlsProperties(namespace, clusterName,
+                caCertName, kafkaUsername, securityProtocol, this.clientId, this.clientSecretName,
+                this.oauthTokenEndpointUri), resultPromise, msgCntPredicate, topicName, clientName));
+
+        try {
+            resultPromise.get(timeoutMs, TimeUnit.MILLISECONDS);
+        } catch (Exception e) {
+            resultPromise.completeExceptionally(e);
+        }
+        vertx.close();
+        return resultPromise;
+    }
+
+    public Future<Integer> receiveMessages(String topicName, String namespace, String clusterName, int messageCount,
+                                           String consumerGroup) throws IOException {
+        return receiveMessages(topicName, namespace, clusterName, messageCount, consumerGroup, Constants.GLOBAL_CLIENTS_TIMEOUT);
+    }
+
+    @Override
+    public Future<Integer> receiveMessages(String topicName, String namespace, String clusterName, int messageCount,
+                                           String consumerGroup, long timeoutMs) throws RuntimeException, IOException {
+        String clientName = "receiver-plain-" + clusterName;
+        vertx = Vertx.vertx();
+        CompletableFuture<Integer> resultPromise = new CompletableFuture<>();
+
+        IntPredicate msgCntPredicate = x -> x == messageCount;
+
+        vertx.deployVerticle(new Consumer(KafkaClientProperties.createOauthConsumerProperties(namespace, clusterName, consumerGroup,
+                this.clientId, this.clientSecretName, this.oauthTokenEndpointUri), resultPromise, msgCntPredicate, topicName, clientName));
+
+        try {
+            resultPromise.get(timeoutMs, TimeUnit.MILLISECONDS);
+        } catch (Exception e) {
+            resultPromise.completeExceptionally(e);
+        }
+        vertx.close();
+        return resultPromise;
+    }
+
+    public Future<Integer> receiveMessagesTls(String topicName, String namespace, String clusterName, String kafkaUsername,
+                                              int messageCount, String securityProtocol, String consumerGroup) throws RuntimeException, IOException {
+        return receiveMessagesTls(topicName, namespace, clusterName, kafkaUsername, messageCount, securityProtocol,
+                consumerGroup, Constants.GLOBAL_CLIENTS_TIMEOUT);
     }
 
     @Override
-    public Future<Integer> receiveMessages(String topicName, String namespace, String clusterName, int messageCount, String consumerGroup) {
-        throw new UnsupportedOperationException();
+    public Future<Integer> receiveMessagesTls(String topicName, String namespace, String clusterName, String kafkaUsername,
```

Comment:
Similarly, how much does this differ from the non-TLS receive? Can you factor out a common method?

Reply:
I have added this note to the existing JIRA, which created Marko. It would take more time and i don't want to make really small quick change.